json_journal/data_analysis.py

```diff
@@ -82,3 +83,49 @@ def total_profit(full_account_name_balances):
             pass
     total = (income * (-1)) - expenses
     return total
+
+
+def cumulative_profit(ledger):
+    '''
+    Calculate the cumulative profit.
+
+    Parameters
+    ----------
+    Ledger : dict
+        Loaded json file.
+
+    Returns
+    -------
+    Dates : list
+        List of dates.
+    Cumulative_profit : list
+        List of cumulative total of profit.
+    '''
+    sorted_ledger = sorted(ledger, key=lambda d: d['date'])
+    dates = []
+    profits = []
+    for transaction in sorted_ledger:
+        transaction_date = transaction['date']
+        entries = transaction['entries']
+        profit = 0
+        for entry in entries:
+            amount = entry['amount']
+            account = entry['account']
+            if (account.startswith("expense:") or account.startswith("income")):
+                profit -= amount
```

Comment:
This makes me a bit nervous. We discussed having some convention around +/- signs and income/expenses. What are our thoughts at the moment?

Reply:
I wasn't too sure how to handle this - previously I had them as separate if statements (one for expense and one for income) but I realised they did the same thing so I combined them. Would it make more sense to separate them back out again and have expense: profit -= amount and income: profit += (-1 * amount)?

Reply:
separated this out - let me know what you think

Reply:
Potentially we separate them into some config file in future. I am not sure how much we should be assuming the program to already know about the input file.